posthog/api/property_definition.py

```diff
@@ -142,13 +146,25 @@ def as_count_sql(self):
         query = f"""
             SELECT count(*) as full_count
             FROM {self.table}
-            left join posthog_eventproperty on posthog_eventproperty.team_id = posthog_propertydefinition.team_id and posthog_eventproperty.property = posthog_propertydefinition.name {self.event_name_join_filter}
+            {self._join_on_event_property()}
             WHERE posthog_propertydefinition.team_id = {self.team_id} AND posthog_propertydefinition.name NOT IN %(excluded_properties)s
              {self.name_filter} {self.numerical_filter} {self.search_query} {self.event_property_filter} {self.is_feature_flag_filter} {self.event_name_filter}
             """
 
         return query
 
+    def _join_on_event_property(self):
+        join_on_posthog_event_property = f"""
```

Comment:
Yuck, should in-line this variable but don't want to lose 20 minutes 🤣 

Will follow-up once we know this fixes the bug